packages/core/data-transfer/lib/providers/local-strapi-destination-provider/restore.ts

```diff
@@ -0,0 +1,47 @@
+import type { ContentTypeSchema } from '@strapi/strapi';
+
+export type DeleteOptions = {
+  contentTypes?: ContentTypeSchema[];
+  uidsOfModelsToDelete?: string[];
+  conditions?: {
+    [contentTypeUid: string]: {
+      params: any;
+    };
+  };
+};
+
+export const deleteAllRecords = async (strapi: Strapi.Strapi, deleteOptions?: DeleteOptions) => {
+  const conditions = deleteOptions?.conditions ?? {};
+  const exceptions = [
```

Comment:
I just find this weird that we're adding default exceptions in the actual provider instead of delegating the logic to the transfer implementations & customization such as the export/import commands.
IMO, the ownership/responsibility for excluding stuff shouldn't be the provider but the engine callers.